frontend/src/pages/ActivityReport/index.js

```diff
@@ -82,25 +83,28 @@ function ActivityReport({ match, user, location }) {
 
   useEffect(() => {
     const fetch = async () => {
+      let report;
+
       try {
         updateLoading(true);
-
-        const apiCalls = [
-          getRecipients(),
-          getCollaborators(region),
-          getApprovers(region),
-        ];
-
         if (activityReportId !== 'new') {
-          apiCalls.push(getReport(activityReportId));
+          report = await getReport(activityReportId);
         } else {
-          apiCalls.push(
-            Promise.resolve({ ...defaultValues, pageState: defaultPageState, userId: user.id }),
-          );
+          report = {
+            ...defaultValues,
+            pageState: defaultPageState,
+            userId: user.id,
+            regionId: region || getRegionWithReadWrite(user),
```

Comment:
`user` and `region` are missing from the dependencies list for `useEffect` One or both of them need to be added depending on whether `region` is really a required property.